proposals/readonly-methods.md

```diff
@@ -0,0 +1,65 @@
+# Readonly Methods
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Provide a way to specify individual methods on a struct do not modify state, in the same way that `readonly struct` specifies no instance method modifies state.
+
+It is worth noting that `readonly method` != `pure method`. A `pure` method guarantees no state will be modified. A `readonly` method only guarantees that instance state will not be modified.
```

Comment:
Changed the big NOTE section I had at the top to this here.

I think it's still worth calling out so that people don't get the wrong idea 😄